### PR TITLE
EX-496 Support debug

### PR DESCRIPTION
**Pull request by robballou**

### CODE REVIEW

File: src/contrib/debug.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import Debug from 'debug';
+import { LoggerInterface, LOG_LEVELS, L } from '../interfaces/logger';
+// import getLogLevelConstant from '../util/logging';
```
Comment (dzrust) [inline src/contrib/debug.ts]: remove if you are not using this

File: src/contrib/debug.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import ContribLogger from './contrib_logger';
+import ContribMetrics from './contrib_metrics';
+import { MetricsInterface } from '../interfaces/metrics';
+import Metric from '../models/metric';
+import { MindfulnessOptions } from '../interfaces/options';
```
Comment (dzrust) [inline src/contrib/debug.ts]: What is Mindfulness?

File: src/contrib/debug.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+   *
+   * This will debug to the namespace property or you can re-use a debug instance
+   * by passing that in as an option.
+   *
+   * @param level The log level to use.
+   * @param message The message or item to log
+   * @param payload Optional additional payload to log
+   * @param options Optional call-specific options for this log.
+   */
+  async call(level: string, message: any, payload?: any, options?: MindfulnessOptions): Promise<any> {
```
Comment (dzrust) [inline src/contrib/debug.ts]: should this be an arrow function?

File: src/contrib/debug.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    return new Promise(async (resolve, reject) => {
+      const callOptions = this.getCallOptions(options);
+      if (typeof callOptions.debugInstance === 'undefined' && !callOptions.namespace) {
+        return reject(new Error('For debug, you must specify either debugInstance or namespace options'));
+      }
+
+      const debugInstance = typeof callOptions.debugInstance !== 'undefined' ? callOptions.debugInstance : Debug(callOptions.namespace);
```
Comment (dzrust) [inline src/contrib/debug.ts]: can this just be: `const debugInstance = !!callOptions.debugInstance ? callOptions.debugInstance : Debug(callOptions.namespace);` or `const debugInstance = !callOptions.debugInstance ?  Debug(callOptions.namespace) : callOptions.debugInstance;`

File: src/contrib/mindfulness.ts [hunk 5]
```diff
@@ -1,5 +1,5 @@
 import { MindfulnessOptions, DetailsInterface } from '../interfaces/options';
-import { Metrics } from '..';
```
Comment (dzrust) [inline src/contrib/mindfulness.ts]: The name of this file does not give insight into what this does

File: test/debug_logger.ts [hunk 6]
```diff
@@ -0,0 +1,10 @@
+import { DebugLogger } from '../src/contrib/debug';
+
+test('ConsoleLogger logs to the console', async (done) => {
```
Comment (dzrust) [inline test/debug_logger.ts]: Do we gain anything from this test being async?

File: test/json_post_logger.ts [hunk 7]
```diff
@@ -15,14 +15,8 @@ test('log via post request to example.com', async (done) => {
...
 
-  const loggingEndpoint = nock('http://logging.example.com')
-    .post('/', {
-      severity: 'log',
-      type: 'log',
-      message: 'Hello!',
-      info: {},
-      environment: 'test',
-    })
+  const loggingEndpoint = nock(/example.com/)
```
Comment (dzrust) [inline test/json_post_logger.ts]: instead of using /example.com/ can you use a bombbomb url for safety

File: test/json_post_logger.ts [hunk 8]
```diff
@@ -36,7 +30,7 @@ test('log via post payload request to example.com', async (done) => {
     { type: 'json_post', host: 'logging.example.com' },
   ]);
 
-  const loggingEndpoint = nock('http://logging.example.com')
+  const loggingEndpoint = nock(/example.com/)
```
Comment (dzrust) [inline test/json_post_logger.ts]: instead of using /example.com/ can you use a bombbomb url for safety

File: test/json_post_logger.ts (hunk 7 above)
Comment (robballou) [reply test/json_post_logger.ts]: example is RFC'd for this purpose

File: test/debug_logger.ts (hunk 6 above)
Comment (robballou) [reply test/debug_logger.ts]: We use await in the test

File: src/contrib/debug.ts (hunk 3 above)
Comment (robballou) [reply src/contrib/debug.ts]: It's an instance method

File: test/json_post_logger.ts (hunk 7 above)
Comment (dzrust) [reply test/json_post_logger.ts]: 👍

File: test/json_post_logger.ts (hunk 8 above)
Comment (robballou) [reply test/json_post_logger.ts]: Example is RFC'd for this purpose plus this library is open source so it's nice not to tie it so directly to our properties.

File: test/debug_logger.ts (hunk 6 above)
Comment (dzrust) [reply test/debug_logger.ts]: :+1:

File: src/contrib/debug.ts (hunk 2 above)
Comment (robballou) [reply src/contrib/debug.ts]: A poor name for the core class but not worth changing for this. I'll make an issue for some changes like that.

**Comment by robballou** [comment]
The dist folder is included so people don't need TS to use the library.

**Pull request merged**